### PR TITLE
Update CMake files to support windows

### PR DESCRIPTION
**Pull request by isuruf**
I've picked some changes from https://github.com/flang-compiler/flang/pull/288 that can be merged easily.

### CODE REVIEW

File: runtime/flang/CMakeLists.txt
Comment (isuruf) [inline runtime/flang/CMakeLists.txt]: This isn't needed because of https://github.com/flang-compiler/flang/pull/464/files#diff-af3b638bc2a3e6c650974192a53c7291R26

File: CMakeLists.txt
Comment (isuruf) [inline CMakeLists.txt]: This is fixed by https://github.com/flang-compiler/flang/pull/464/files#diff-af3b638bc2a3e6c650974192a53c7291R30

**Comment by isuruf** [comment]
@sscalpone, @gklimowicz, this PR is ready for review. This does not add support for Windows, but adds a few miscellaneous changes to the CMake files that can be reviewed independently. (I figured small PRs are easier to review.)

File: CMakeLists.txt
Comment (cbezault) [inline CMakeLists.txt]: While Flang might in general not be capable of being cross-compiled (I have no idea either way) these two lines are definitely cross-compile unfriendly.

Comment (isuruf) [reply CMakeLists.txt]: Yes, this should be `CMAKE_SYSTEM_NAME` instead of `CMAKE_HOST_SYSTEM_NAME`

**Comment by kiranchandramohan** [comment]
@isuruf Were the Pull requests for Windows finally merged into flang or they are still not merged like this one? What is the status of current flang master for Windows and what will be the status if we merge all the Windows patches?

**Comment by isuruf** [comment]
Only a couple were merged. See https://github.com/flang-compiler/flang/pulls?q=is%3Apr+author%3Aisuruf+

I haven't checked in a long time, but the pull request #464 was the original PR and I broke it into smaller PRs, but not all of the changes of #464 were broken into smaller PRs. After the other PRs are done, will have to check what needs to be ported over from #464

**Comment by isuruf** [comment]
I meant #288 above. Edited the comment

**Comment by kaadam** [comment]
HI @isuruf, I'm interested in Windows support of flang. What's your plan with it? Do you want to merge these patch set? 
Have you worked on Windows? Or used mingw?

**Comment by kiranchandramohan** [comment]
@isuruf We now have commit access and PRs are being merged as you might have noticed. We can process this and other Windows PRs if they are ready and if there is interest from others.

There is also a biweekly call discussing PRs and Classic Flang in general. Minutes and teams meeting link can be found in the google doc below.
https://docs.google.com/document/d/1-OuiKx4d7O6eLEJDBDKSRnSiUO2rgRR-c2Ga4AkrzOI

**Comment by isuruf** [comment]
@kiranchandramohan, that'd be great. This PR is ready and so are the other PRs that I have authored (https://github.com/flang-compiler/flang/pulls/isuruf). Merging all of those would be the first step towards Windows support. There's more work to do after these PRs are merged.

**Comment by kiranchandramohan** [comment]
Can you rebase these PRs and also give the order in which these should be tested?

**Comment by kiranchandramohan** [comment]
@isuruf the github action CI is enabled now, it creates a build on X86 linux and does make check-all. Would you be able to check and edit your PR so that the CI passes?

**Comment by isuruf** [comment]
Yes, will do. I'm talking with CMake developers about how to fix this in https://gitlab.kitware.com/cmake/cmake/-/issues/21463

**Comment by seshakalyur** [comment]
Hello, 
I am interested in getting access to Flang on Windows.  What is the the latest status? Are we there yet?
Thanks,

**Comment by kaadam** [comment]
> Yes, will do. I'm talking with CMake developers about how to fix this in https://gitlab.kitware.com/cmake/cmake/-/issues/21463

@isuruf Hi, What about your CMake fix? I saw you already have a WIP patch for that (at the moment the PR is closed), are you going to merge this fix?

**Comment by seshakalyur** [comment]
Hello All,
I am interested in finding out where windows support for flang builds stands.
Pls. advise.
Thanks,
-Sesha

**Comment by kiranchandramohan** [comment]
@seshakalyur We are in the preliminary stages of Windows support. The first task is to enable CMake support using patches from @isuruf so that we can start compiling the source of the compiler. Then we will look at the source compilation issues. I believe @kaadam has taken this CMake changes and probably others and got the CMake part working.

File: runtime/flang/CMakeLists.txt
Comment (kaadam) [inline runtime/flang/CMakeLists.txt:11]: Maybe it's better to check WIN32 instead of MSVC. We could use clang with regular front end on Windows which detected as GNU compiler.

File: runtime/flang/CMakeLists.txt
Comment (kaadam) [inline runtime/flang/CMakeLists.txt:593]: Same as above

File: runtime/flangrti/CMakeLists.txt
Comment (kaadam) [inline runtime/flangrti/CMakeLists.txt:70]: Same as above

**Comment by kaadam** [comment]
@isuruf @kiranchandramohan What about if intrinsic keywords are removed as a workaround from ieee_arithmetic.F95 and ieee_exceptions.F95?

**Comment by kaadam** [comment]
@isuruf ping?

File: CMakeLists.txt
Comment (bryanpkc) [inline CMakeLists.txt:14]: Do you run CMake with `-DCMAKE_Fortran_COMPILER_ID=Flang`? That should set `CMAKE_Fortran_MODDIR_FLAG` correctly.

File: runtime/flang/CMakeLists.txt
Comment (bryanpkc) [inline runtime/flang/CMakeLists.txt:489]: Please indent the two `set_property` lines.

File: runtime/flangrti/CMakeLists.txt
Comment (bryanpkc) [reply runtime/flangrti/CMakeLists.txt:70]: Can we teach the driver to look for `flangrti.dll` instead of `libflangrti.dll`?

File: runtime/ompstub/CMakeLists.txt
Comment (bryanpkc) [inline runtime/ompstub/CMakeLists.txt:13]: Please indent the two `set_property` lines.

File: CMakeLists.txt
Comment (isuruf) [reply CMakeLists.txt:14]: It doesn't for me. (at least on windows)

File: runtime/flangrti/CMakeLists.txt
Comment (isuruf) [reply runtime/flangrti/CMakeLists.txt:65]: In MSVC, the structure of the files is like
- flang.lib  - import library for dynamic library
- flang.dll  - dynamic library
- libflang.lib - static library

Notice that the import library and the static library have the same extension and hence the need for two different names.

For MinGW,
- libflang.dll.a - import library for dynamic library
- libflang.dll - dynamic library
- libflang.a  - static library

Comment (bryanpkc) [reply runtime/flangrti/CMakeLists.txt:65]: I see, life is complicated. Thanks.

**Review by kaadam — APPROVED** [review]
LGTM, (I will close #1003 )

File: CMakeLists.txt
Comment (bryanpkc) [reply CMakeLists.txt:14]: I'm still wondering why these are needed. On Linux, during a clean build (from scratch), `flang` is not yet available or usable, CMake reports the following:
```
-- Detecting Fortran compiler ABI info
-- Detecting Fortran compiler ABI info - failed
-- Checking whether /home/b00375952/src/install.release_12x/bin/flang supports Fortran 90
-- Checking whether /home/b00375952/src/install.release_12x/bin/flang supports Fortran 90 - no
```
But it continues on to build the Flang frontend and runtime just fine. What error do you see when building on Windows, if you don't add these lines to `CMakeLists.txt`?

Comment (isuruf) [reply CMakeLists.txt:14]:
```
CMake Error at C:/Users/Administrator/miniconda3/envs/llvm/Library/share/cmake-3.20/Modules/CMakeDetermineCompilerABI.cmake:49 (try_compile):
  Failed to generate test project build system.
Call Stack (most recent call first):
  C:/Users/Administrator/miniconda3/envs/llvm/Library/share/cmake-3.20/Modules/CMakeTestFortranCompiler.cmake:20 (CMAKE_DETERMINE_COMPILER_ABI)
  runtime/flang/CMakeLists.txt:7 (enable_language)
  ```

Comment (isuruf) [reply CMakeLists.txt:14]: If second line is not there
```
CMake Error: Error required internal CMake variable not set, cmake may not be built correctly.
Missing variable is:
CMAKE_Fortran_PREPROCESS_SOURCE
```

Comment (isuruf) [reply CMakeLists.txt:14]: If third line is not there, the mod files are not installed to the correct location.

Comment (isuruf) [reply CMakeLists.txt]: I removed the last 2 lines because they are actually set in `runtime/flang/CMakeLists.txt` when `enable_languages(Fortran)` is called.
The first line is still needed.

Comment (bryanpkc) [reply CMakeLists.txt]: Can you make CMakeError.log available? I believe that the root cause is that `CMAKE_DETERMINE_COMPILER_ABI` cannot find the `flang` executable, which should have been installed when you built and installed classic-flang-llvm-project, and should be reachable via your `PATH` environment variable. The `flang` executable is a symbolic link to `clang` on Linux; I am not sure what form it takes on Windows. If it can be found and executed, CMake should be able to figure out other information about the Flang compiler, and continue without these hacks.

Comment (isuruf) [reply CMakeLists.txt]:
```
Compiling the Fortran compiler identification source file "CMakeFortranCompilerId.F" failed.
Compiler: C:/Users/Administrator/miniconda3/envs/llvm/Library/bin/flang.exe 
Build flags: 
Id flags: -v 

The output was:
1
clang version 10.0.1 
Target: x86_64-pc-windows-msvc
Thread model: posix
InstalledDir: C:/Users/Administrator/miniconda3/envs/llvm/Library/bin
 "flang1" CMakeFortranCompilerId.F -opt 0 -terse 1 -inform warn -nohpf -nostatic -cmdline "'+flang -v CMakeFortranCompilerId.F'" -inform warn -ieee 0 -x 19 0x400000 -quad -x 68 0x1 -x 59 4 -x 15 2 -x 49 0x400004 -x 51 0x20 -x 57 0x4c -x 58 0x10000 -x 124 0x1000 -tp px -x 57 0xfb0000 -x 58 0x78031040 -x 47 0x08 -x 48 4608 -x 49 0x100 -def __GCC_ASM_FLAG_OUTPUTS__=1 -def __code_model__=1 -def __amd64__=1 -def __amd64=1 -def __x86_64=1 -def __x86_64__=1 -def __SEG_GS=1 -def __SEG_FS=1 -def __seg_gs=__attribute__((address_space(256))) -def __seg_fs=__attribute__((address_space(257))) -def __REGISTER_PREFIX__= -def __NO_MATH_INLINES=1 -def __SSE2__=1 -def __SSE2_MATH__=1 -def __SSE__=1 -def __SSE_MATH__=1 -def __MMX__=1 -def __GCC_HAVE_SYNC_COMPARE_AND_SWAP_1=1 -def __GCC_HAVE_SYNC_COMPARE_AND_SWAP_2=1 -def __GCC_HAVE_SYNC_COMPARE_AND_SWAP_4=1 -def __GCC_HAVE_SYNC_COMPARE_AND_SWAP_8=1 -def _WIN32=1 -def _WIN64=1 -def _MSC_VER=1916 -def _MSC_FULL_VER=191627043 -def _MSC_BUILD=1 -def _INTEGRAL_MAX_BITS=64 -def _M_X64=100 -def _M_AMD64=100 -def "__SIZE_TYPE__=long long unsigned int" -def "__PTRDIFF_TYPE__=long long int" -def __LONG_MAX__=2147483647L -def __amd_64__amd64__ -def __k8 -def __k8__ -def __THROW= -def __extension__= -def __PGLLVM__ -preprocess -nofreeform -vect 48 -x 54 1 -x 70 0x40000000 -y 163 0xc0000000 -x 189 0x10 -stbfile CMakeFortranCompilerId-538e6f.stb -modexport CMakeFortranCompilerId-538e6f.cmod -modindex CMakeFortranCompilerId-538e6f.cmdx -output CMakeFortranCompilerId-538e6f.ilm
flang: error: unable to execute command: program not executable
```

Comment (bryanpkc) [reply CMakeLists.txt]: Odd. On Linux I get these failures too but CMake still manages to make progress:

```
Compiling the Fortran compiler identification source file "CMakeFortranCompilerId.F" failed.
Compiler: /home/b00375952/src/install.release_12x/bin/flang
Build flags:
Id flags: -v

The output was:
1
flang-new version 12.0.0 (https://github.com/flang-compiler/classic-flang-llvm-project d9e7b49409b89afb32142ba67f3f1aa0d8479a98)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /home/b00375952/src/install.release_12x/bin
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/5
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/5.5.0
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/6
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/6.4.0
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/7
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/7.5.0
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/8
Selected GCC installation: /usr/lib/gcc/x86_64-linux-gnu/7.5.0
Candidate multilib: .;@m64
Selected multilib: .;@m64
 "flang1" CMakeFortranCompilerId.F -opt 0 -terse 1 -inform warn -nohpf -nostatic -cmdline "'+flang -v CMakeFortranCompilerId.F'" -inform warn -ieee 0 -x 19 0x400000 -quad -x 68 0x1 -x 59 4 -x 15 2 -x 49 0x400004 -x 51 0x20 -x 57 0x4c -x 58 0x10000 -x 124 0x1000 -tp px -x 57 0xfb0000 -x 58 0x78031040 -x 47 0x08 -x 48 4608 -x 49 0x100 -stdinc /home/b00375952/src/install.release_12x/bin/../include:/usr/local/include:/home/b00375952/src/install.release_12x/lib/clang/12.0.0/include:/usr/include/x86_64-linux-gnu:/include:/usr/include -def __GCC_ASM_FLAG_OUTPUTS__=1 -def __code_model___=1 -def __amd64__=1 -def __amd64=1 -def __x86_64=1 -def __x86_64__=1 -def __SEG_GS=1 -def __SEG_FS=1 -def __seg_gs=__attribute__((address_space(256))) -def __seg_fs=__attribute__((address_space(257))) -def __REGISTER_PREFIX__= -def __NO_MATH_INLINES=1 -def __SSE2__=1 -def __SSE2_MATH__=1 -def __SSE__=1 -def __SSE_MATH__=1 -def __MMX__=1 -def __GCC_HAVE_SYNC_COMPARE_AND_SWAP_1=1 -def __GCC_HAVE_SYNC_COMPARE_AND_SWAP_2=1 -def __GCC_HAVE_SYNC_COMPARE_AND_SWAP_4=1 -def __GCC_HAVE_SYNC_COMPARE_AND_SWAP_8=1 -def __SIZEOF_FLOAT128__=16 -def unix=1 -def __unix=1 -def __unix__=1 -def linux=1 -def __linux=1 -def __linux__=1 -def __ELF__=1 -def __gnu_linux__=1 -def __FLOAT128__=1 -def "__SIZE_TYPE__=long unsigned int" -def "__PTRDIFF_TYPE__=long int" -def _LP64=1 -def __LP64__=1 -def __LONG_MAX__=9223372036854775807L -def __amd_64__amd64__ -def __k8 -def __k8__ -def __THROW= -def __extension__= -def __PGLLVM__ -preprocess -nofreeform -vect 48 -x 54 1 -x 70 0x40000000 -y 163 0xc0000000 -x 189 0x10 -stbfile CMakeFortranCompilerId-bce089.stb -modexport CMakeFortranCompilerId-bce089.cmod -modindex CMakeFortranCompilerId-bce089.cmdx -output CMakeFortranCompilerId-bce089.ilm
clang-12: error: unable to execute command: Executable "flang1" doesn't exist!


Detecting Fortran compiler ABI info failed to compile with the following output:
Change Dir: /home/b00375952/src/build.release_12x/flang/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/bin/make cmTC_3b3e4/fast && /usr/bin/make  -f CMakeFiles/cmTC_3b3e4.dir/build.make CMakeFiles/cmTC_3b3e4.dir/build
make[1]: Entering directory '/home/b00375952/src/build.release_12x/flang/CMakeFiles/CMakeTmp'
Building Fortran object CMakeFiles/cmTC_3b3e4.dir/CMakeFortranCompilerABI.F.o
/home/b00375952/src/install.release_12x/bin/flang   -v -c /usr/share/cmake-3.19/Modules/CMakeFortranCompilerABI.F -o CMakeFiles/cmTC_3b3e4.dir/CMakeFortranCompilerABI.F.o
flang-new version 12.0.0 (https://github.com/flang-compiler/classic-flang-llvm-project d9e7b49409b89afb32142ba67f3f1aa0d8479a98)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /home/b00375952/src/install.release_12x/bin
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/5
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/5.5.0
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/6
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/6.4.0
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/7
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/7.5.0
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/8
Selected GCC installation: /usr/lib/gcc/x86_64-linux-gnu/7.5.0
Candidate multilib: .;@m64
Selected multilib: .;@m64
 "flang1" /usr/share/cmake-3.19/Modules/CMakeFortranCompilerABI.F -opt 0 -terse 1 -inform warn -nohpf -nostatic -cmdline "'+flang -v -c /usr/share/cmake-3.19/Modules/CMakeFortranCompilerABI.F -o CMakeFiles/cmTC_3b3e4.dir/CMakeFortranCompilerABI.F.o'" -inform warn -ieee 0 -x 19 0x400000 -quad -x 68 0x1 -x 59 4 -x 15 2 -x 49 0x400004 -x 51 0x20 -x 57 0x4c -x 58 0x10000 -x 124 0x1000 -tp px -x 57 0xfb0000 -x 58 0x78031040 -x 47 0x08 -x 48 4608 -x 49 0x100 -stdinc /home/b00375952/src/install.release_12x/bin/../include:/usr/local/include:/home/b00375952/src/install.release_12x/lib/clang/12.0.0/include:/usr/include/x86_64-linux-gnu:/include:/usr/include -def __GCC_ASM_FLAG_OUTPUTS__=1 -def __code_model___=1 -def __amd64__=1 -def __amd64=1 -def __x86_64=1 -def __x86_64__=1 -def __SEG_GS=1 -def __SEG_FS=1 -def __seg_gs=__attribute__((address_space(256))) -def __seg_fs=__attribute__((address_space(257))) -def __REGISTER_PREFIX__= -def __NO_MATH_INLINES=1 -def __SSE2__=1 -def __SSE2_MATH__=1 -def __SSE__=1 -def __SSE_MATH__=1 -def __MMX__=1 -def __GCC_HAVE_SYNC_COMPARE_AND_SWAP_1=1 -def __GCC_HAVE_SYNC_COMPARE_AND_SWAP_2=1 -def __GCC_HAVE_SYNC_COMPARE_AND_SWAP_4=1 -def __GCC_HAVE_SYNC_COMPARE_AND_SWAP_8=1 -def __SIZEOF_FLOAT128__=16 -def unix=1 -def __unix=1 -def __unix__=1 -def linux=1 -def __linux=1 -def __linux__=1 -def __ELF__=1 -def __gnu_linux__=1 -def __FLOAT128__=1 -def "__SIZE_TYPE__=long unsigned int" -def "__PTRDIFF_TYPE__=long int" -def _LP64=1 -def __LP64__=1 -def __LONG_MAX__=9223372036854775807L -def __amd_64__amd64__ -def __k8 -def __k8__ -def __THROW= -def __extension__= -def __PGLLVM__ -preprocess -nofreeform -vect 48 -x 54 1 -x 70 0x40000000 -y 163 0xc0000000 -x 189 0x10 -stbfile CMakeFortranCompilerABI-0004c9.stb -modexport CMakeFortranCompilerABI-0004c9.cmod -modindex CMakeFortranCompilerABI-0004c9.cmdx -output CMakeFortranCompilerABI-0004c9.ilm
clang-12: error: unable to execute command: Executable "flang1" doesn't exist!
CMakeFiles/cmTC_3b3e4.dir/build.make:84: recipe for target 'CMakeFiles/cmTC_3b3e4.dir/CMakeFortranCompilerABI.F.o' failed
make[1]: *** [CMakeFiles/cmTC_3b3e4.dir/CMakeFortranCompilerABI.F.o] Error 1
make[1]: Leaving directory '/home/b00375952/src/build.release_12x/flang/CMakeFiles/CMakeTmp'
Makefile:140: recipe for target 'cmTC_3b3e4/fast' failed
make: *** [cmTC_3b3e4/fast] Error 2
```
What happens if you add `-DCMAKE_Fortran_COMPILER_WORKS=ON` when invoking CMake?

Comment (isuruf) [reply CMakeLists.txt]: Doesn't work either

Comment (bryanpkc) [reply CMakeLists.txt]: OK, I'll rest my case for now. 😄

File: CMakeLists.txt
Comment (isuruf) [inline CMakeLists.txt:20]: @bryanpkc, both of these are needed. (I accidentally had the second in my command line when I said it's not needed.)
Without the second line it fails with,
```
CMake Error at C:/Users/Administrator/miniconda3/envs/llvm/Library/share/cmake-3.20/Modules/Platform/Windows-MSVC.cmake:71 (message):
  MSVC compiler version not detected properly:
Call Stack (most recent call first):
  C:/Users/Administrator/miniconda3/envs/llvm/Library/share/cmake-3.20/Modules/Platform/Windows-Flang-Fortran.cmake:1 (include)
  C:/Users/Administrator/miniconda3/envs/llvm/Library/share/cmake-3.20/Modules/CMakeFortranInformation.cmake:25 (include)
  C:/Users/isuru/projects/flang/build/CMakeFiles/CMakeTmp/CMakeLists.txt:4 (project)


CMake Error at C:/Users/Administrator/miniconda3/envs/llvm/Library/share/cmake-3.20/Modules/CMakeTestFortranCompiler.cmake:70 (try_compile):
  Failed to configure test project build system.
Call Stack (most recent call first):
  runtime/flang/CMakeLists.txt:7 (enable_language)


-- Configuring incomplete, errors occurred!
```

Comment (bryanpkc) [reply CMakeLists.txt:20]: OK, no problem.

File: CMakeLists.txt
Comment (bryanpkc) [inline CMakeLists.txt:13]: If this is absolutely needed here, near the top of the file, then I suggest removing lines 59-60.

Comment (isuruf) [reply CMakeLists.txt:13]: Done in a4c90f89

File: CMakeLists.txt
Comment (bryanpkc) [inline CMakeLists.txt:34]: Does the Windows build always print this message and call `return()`?

Comment (isuruf) [reply CMakeLists.txt:34]: Yes, I didn't want to call `return()` until the port was complete. I can remove it if you prefer.

Comment (bryanpkc) [reply CMakeLists.txt:34]: Oh I see. OK.

**Comment by isuruf** [comment]
Anything else to do here?

**Comment by kaadam** [comment]
@bryanpkc do you have any other comment on this PR? Could you look at it please? Thanks in advance.

**Review by bryanpkc — CHANGES_REQUESTED** [review]
I don't have any more comment on this PR, except that the commits should be squashed and a comprehensive commit message should be written for the squashed commit.

**Comment by isuruf** [comment]
I squashed the commits into two commits. One for all the commits authored by me and one for all the commits by Albert.

File: runtime/flang/CMakeLists.txt
Comment (pawosm-arm) [reply runtime/flang/CMakeLists.txt:11]: ...but, is the linker issue windows specific or rather MSVC specific?

File: runtime/flangrti/CMakeLists.txt
Comment (pawosm-arm) [inline runtime/flangrti/CMakeLists.txt:79]: See also MSVC question above (in the runtime/flang/CMakeLists.txt file)

File: tools/flang2/flang2exe/CMakeLists.txt
Comment (pawosm-arm) [inline tools/flang2/flang2exe/CMakeLists.txt:146]: See also MSVC comment above (runtime/flang/CMakeLists.txt)

File: runtime/flang/CMakeLists.txt
Comment (pawosm-arm) [reply runtime/flang/CMakeLists.txt:662]: Same as in the other commit of this refactored PR.

File: runtime/flang/CMakeLists.txt
Comment (isuruf) [reply runtime/flang/CMakeLists.txt:11]: All of these are MSVC specific and doesn't affect MinGW.
CMake will identify clang with target triple `x86_64-windows-msvc` as MSVC and clang with target triple `x86_64-w64-mingw32` as MinGW.

**Review by pawosm-arm — APPROVED** [review]
LGTM, although I'd feel better if an Windows expert could also approve that.

**Review by bryanpkc — APPROVED** [review]
LGTM.

**Pull request merged**
